ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -360,14 +360,24 @@ extension SignalProducerType {
 		return start(Observer(observerAction))
 	}
 
-	/// Creates a Signal from the producer, then adds exactly one observer to
-	/// the Signal, which will invoke the given callback when `next` events are
-	/// received.
+	@available(*, deprecated=1.0, message="This SignalProducer may emit errors which must be handled explicitly, or observed using startWithResult:")
+	public func startWithNext(next: Value -> Void) -> Disposable {
+		fatalError("Unavailable method")
+	}
+
+	/// Creates a Signal from the producer, then adds an observer to
+	/// the Signal, which will invoke the given callback when `next` or `failed` 
+	/// events are received.
 	///
 	/// Returns a Disposable which can be used to interrupt the work associated
 	/// with the Signal, and prevent any future callbacks from being invoked.
-	public func startWithNext(next: Value -> Void) -> Disposable {
-		return start(Observer(next: next))
+	public func startWithResult(result: Result<Value, Error> -> Void) -> Disposable {
```

Comment:
I feel like using the singular "result" here could be a little misleading. It almost sounds too final, like there will only be one. What about `.startWithResults {}`? I do think this could use a little bikeshedding, though.